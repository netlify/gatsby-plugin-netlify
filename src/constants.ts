// Gatsby values
export const BUILD_BROWSER_BUNDLE_STAGE = `build-javascript`

// Plugin values
export const NETLIFY_HEADERS_FILENAME = `_headers`

// Default options including transform to manipulate headers for
// sorting and rewrites for client only paths
export const DEFAULT_OPTIONS = {
  headers: {},
  mergeSecurityHeaders: true,
  mergeLinkHeaders: true,
  mergeCachingHeaders: true,
  transformHeaders: (value) => value,
  generateMatchPathRewrites: true,
}

export const SECURITY_HEADERS = {
  '/*': [
    `X-Frame-Options: DENY`,
    `X-XSS-Protection: 1; mode=block`,
    `X-Content-Type-Options: nosniff`,
    `Referrer-Policy: same-origin`,
  ],
}

export const IMMUTABLE_CACHING_HEADER = `Cache-Control: public, max-age=31536000, immutable`

export const CACHING_HEADERS = {
  '/static/*': [IMMUTABLE_CACHING_HEADER],
  '/sw.js': [`Cache-Control: no-cache`],
}

export const LINK_REGEX = /^(Link: <\/)(.+)(>;.+)/
export const ROOT_WILDCARD = `/*`

export const HEADER_COMMENT = `## Created with gatsby-plugin-netlify`

export const PAGE_COUNT_WARN = 1000
