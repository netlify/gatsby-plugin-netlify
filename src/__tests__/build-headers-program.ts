import { existsSync, readFile } from 'fs-extra'

import buildHeadersProgram from '../build-headers-program'
import { DEFAULT_OPTIONS } from '../constants'

import { createPluginData } from './helpers'

jest.mock(`fs-extra`, () => {
  const actualFsExtra = jest.requireActual(`fs-extra`)
  return {
    ...actualFsExtra,
    existsSync: jest.fn(),
  }
})
// eslint-disable-next-line max-lines-per-function
describe(`build-headers-program`, () => {
  let reporter: any

  beforeEach(() => {
    reporter = {
      panic: jest.fn(),
    }
    existsSync.mockClear()
    existsSync.mockReturnValue(true)
  })

  it(`with manifest['pages-manifest']`, async () => {
    const pluginData = await createPluginData()

    existsSync.mockImplementation((path: any) => !path.includes(`page-data.json`) && !path.includes(`app-data.json`))

    // gatsby < 2.9 uses page-manifest
    pluginData.manifest[`pages-manifest`] = [`pages-manifest-ab11f09e0ca7ecd3b43e.js`]

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      mergeCachingHeaders: true,
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    const output = await readFile(pluginData.publicFolder(`_headers`), `utf8`)
    expect(output).toMatchSnapshot()
    expect(output).toMatch(/\/pages-manifest-ab11f09e0ca7ecd3b43e\.js/g)
    expect(output).not.toMatch(/\/undefined/g)
  })

  it(`without app-data file`, async () => {
    const pluginData = await createPluginData()

    // gatsby 2.17.0+ adds an app-data file
    delete pluginData.manifest[`pages-manifest`]

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      mergeCachingHeaders: true,
    }
    existsSync.mockImplementation((path: any) => !path.includes(`app-data.json`))

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    const output = await readFile(pluginData.publicFolder(`_headers`), `utf8`)
    expect(output).not.toMatch(/app-data\.json/g)
    expect(output).not.toMatch(/\/undefined/g)
  })

  it(`without caching headers`, async () => {
    const pluginData = await createPluginData()

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      mergeCachingHeaders: false,
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    expect(await readFile(pluginData.publicFolder(`_headers`), `utf8`)).toMatchSnapshot()
  })

  it(`with security headers`, async () => {
    const pluginData = await createPluginData()

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      mergeSecurityHeaders: true,
      headers: {
        '/*': [
          `Content-Security-Policy: frame-ancestors 'self' https://*.storyblok.com/`,
          `X-Frame-Options: ALLOW-FROM https://app.storyblok.com/`,
        ],
        '/hello': [`X-Frame-Options: SAMEORIGIN`],
      },
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    expect(await readFile(pluginData.publicFolder(`_headers`), `utf8`)).toMatchSnapshot()
  })

  it(`with badly headers configuration`, async () => {
    const pluginData = await createPluginData()

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      mergeSecurityHeaders: true,
      headers: {
        'X-Frame-Options': [`sameorigin`],
      },
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).toHaveBeenCalled()
  })

  it(`with all page headers configuration`, async () => {
    const pluginData = await createPluginData()

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      allPageHeaders: [
        `X-Custom-PageHeader: SomeCustomValue`
      ]
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    expect(await readFile(pluginData.publicFolder(`_headers`), `utf8`)).toMatchSnapshot()
  })  

  it(`with transform headers configuration`, async () => {
    const pluginData = await createPluginData()

    const pluginOptions = {
      ...DEFAULT_OPTIONS,
      transformHeaders: (headers, path) => {
        if (path !== '/webpack-runtime-acaa8994f1f704475e21.js') {
          return headers
        }        

        headers.push('X-TransformedPageHeader: webpack-runtime')
        return headers
      }
    }

    await buildHeadersProgram(pluginData, pluginOptions, reporter)

    expect(reporter.panic).not.toHaveBeenCalled()
    expect(await readFile(pluginData.publicFolder(`_headers`), `utf8`)).toMatchSnapshot()
  })
})
