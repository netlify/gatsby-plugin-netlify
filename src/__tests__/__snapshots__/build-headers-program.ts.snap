// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build-headers-program with caching headers 1`] = `
"## Created with gatsby-plugin-netlify

/*
  X-Frame-Options: DENY
  X-XSS-Protection: 1; mode=block
  X-Content-Type-Options: nosniff
  Referrer-Policy: same-origin
/component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js
  Cache-Control: public, max-age=31536000, immutable
/0-0180cd94ef2497ac7db8.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-templates-blog-post-js-517987eae96e75cddbe7.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-404-js-53e6c51a5a7e73090f50.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-index-js-0bdd01c77ee09ef0224c.js
  Cache-Control: public, max-age=31536000, immutable
/webpack-runtime-acaa8994f1f704475e21.js
  Cache-Control: public, max-age=31536000, immutable
/styles.1025963f4f2ec7abbad4.css
  Cache-Control: public, max-age=31536000, immutable
/styles-565f081c8374bbda155f.js
  Cache-Control: public, max-age=31536000, immutable
/app-f33c13590352da20930f.js
  Cache-Control: public, max-age=31536000, immutable
/static/*
  Cache-Control: public, max-age=31536000, immutable
/sw.js
  Cache-Control: no-cache
/offline-plugin-app-shell-fallback/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/offline-plugin-app-shell-fallback/page-data.json>; rel=preload; as=fetch; crossorigin
/hi-folks/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hi-folks/page-data.json>; rel=preload; as=fetch; crossorigin
/my-second-post/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/my-second-post/page-data.json>; rel=preload; as=fetch; crossorigin
/hello-world/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hello-world/page-data.json>; rel=preload; as=fetch; crossorigin
/404/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404/page-data.json>; rel=preload; as=fetch; crossorigin
/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-index-js-0bdd01c77ee09ef0224c.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/index/page-data.json>; rel=preload; as=fetch; crossorigin
/404.html
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404.html/page-data.json>; rel=preload; as=fetch; crossorigin
"
`;

exports[`build-headers-program with manifest['pages-manifest'] 1`] = `
"## Created with gatsby-plugin-netlify

/*
  X-Frame-Options: DENY
  X-XSS-Protection: 1; mode=block
  X-Content-Type-Options: nosniff
  Referrer-Policy: same-origin
/component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js
  Cache-Control: public, max-age=31536000, immutable
/0-0180cd94ef2497ac7db8.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-templates-blog-post-js-517987eae96e75cddbe7.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-404-js-53e6c51a5a7e73090f50.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-index-js-0bdd01c77ee09ef0224c.js
  Cache-Control: public, max-age=31536000, immutable
/pages-manifest-ab11f09e0ca7ecd3b43e.js
  Cache-Control: public, max-age=31536000, immutable
/webpack-runtime-acaa8994f1f704475e21.js
  Cache-Control: public, max-age=31536000, immutable
/styles.1025963f4f2ec7abbad4.css
  Cache-Control: public, max-age=31536000, immutable
/styles-565f081c8374bbda155f.js
  Cache-Control: public, max-age=31536000, immutable
/app-f33c13590352da20930f.js
  Cache-Control: public, max-age=31536000, immutable
/static/*
  Cache-Control: public, max-age=31536000, immutable
/sw.js
  Cache-Control: no-cache
/offline-plugin-app-shell-fallback/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js>; rel=preload; as=script
/hi-folks/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
/my-second-post/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
/hello-world/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
/404/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-index-js-0bdd01c77ee09ef0224c.js>; rel=preload; as=script
/404.html
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
"
`;

exports[`build-headers-program with security headers 1`] = `
"## Created with gatsby-plugin-netlify

/*
  X-Frame-Options: ALLOW-FROM https://app.storyblok.com/
  X-XSS-Protection: 1; mode=block
  X-Content-Type-Options: nosniff
  Referrer-Policy: same-origin
  Content-Security-Policy: frame-ancestors 'self' https://*.storyblok.com/
/hello
  X-Frame-Options: SAMEORIGIN
/component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js
  Cache-Control: public, max-age=31536000, immutable
/0-0180cd94ef2497ac7db8.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-templates-blog-post-js-517987eae96e75cddbe7.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-404-js-53e6c51a5a7e73090f50.js
  Cache-Control: public, max-age=31536000, immutable
/component---src-pages-index-js-0bdd01c77ee09ef0224c.js
  Cache-Control: public, max-age=31536000, immutable
/webpack-runtime-acaa8994f1f704475e21.js
  Cache-Control: public, max-age=31536000, immutable
/styles.1025963f4f2ec7abbad4.css
  Cache-Control: public, max-age=31536000, immutable
/styles-565f081c8374bbda155f.js
  Cache-Control: public, max-age=31536000, immutable
/app-f33c13590352da20930f.js
  Cache-Control: public, max-age=31536000, immutable
/static/*
  Cache-Control: public, max-age=31536000, immutable
/sw.js
  Cache-Control: no-cache
/offline-plugin-app-shell-fallback/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/offline-plugin-app-shell-fallback/page-data.json>; rel=preload; as=fetch; crossorigin
/hi-folks/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hi-folks/page-data.json>; rel=preload; as=fetch; crossorigin
/my-second-post/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/my-second-post/page-data.json>; rel=preload; as=fetch; crossorigin
/hello-world/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hello-world/page-data.json>; rel=preload; as=fetch; crossorigin
/404/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404/page-data.json>; rel=preload; as=fetch; crossorigin
/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-index-js-0bdd01c77ee09ef0224c.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/index/page-data.json>; rel=preload; as=fetch; crossorigin
/404.html
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404.html/page-data.json>; rel=preload; as=fetch; crossorigin
"
`;

exports[`build-headers-program without caching headers 1`] = `
"## Created with gatsby-plugin-netlify

/*
  X-Frame-Options: DENY
  X-XSS-Protection: 1; mode=block
  X-Content-Type-Options: nosniff
  Referrer-Policy: same-origin
/offline-plugin-app-shell-fallback/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </component---node-modules-gatsby-plugin-offline-app-shell-js-78f9e4dea04737fa062d.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/offline-plugin-app-shell-fallback/page-data.json>; rel=preload; as=fetch; crossorigin
/hi-folks/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hi-folks/page-data.json>; rel=preload; as=fetch; crossorigin
/my-second-post/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/my-second-post/page-data.json>; rel=preload; as=fetch; crossorigin
/hello-world/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-templates-blog-post-js-517987eae96e75cddbe7.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/hello-world/page-data.json>; rel=preload; as=fetch; crossorigin
/404/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404/page-data.json>; rel=preload; as=fetch; crossorigin
/
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-index-js-0bdd01c77ee09ef0224c.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/index/page-data.json>; rel=preload; as=fetch; crossorigin
/404.html
  Link: </webpack-runtime-acaa8994f1f704475e21.js>; rel=preload; as=script
  Link: </styles-565f081c8374bbda155f.js>; rel=preload; as=script
  Link: </app-f33c13590352da20930f.js>; rel=preload; as=script
  Link: </0-0180cd94ef2497ac7db8.js>; rel=preload; as=script
  Link: </component---src-pages-404-js-53e6c51a5a7e73090f50.js>; rel=preload; as=script
  Link: </page-data/app-data.json>; rel=preload; as=fetch; crossorigin
  Link: </page-data/404.html/page-data.json>; rel=preload; as=fetch; crossorigin
"
`;
